packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts

```diff
@@ -443,18 +437,36 @@ export class DotNetGenerator extends Generator {
         }
     }
 
-    // TODO: I uncovered an issue with optional parameters and ordering them
-    // They are currently not supported by the generator.
-    // In C#, optional parameters need to be declared after required parameters
-    // We could make changes to the parameters ordering in the jsii model to support them
-    // But then an optional parameter becoming non optional would create a mess
-    // https://github.com/awslabs/jsii/issues/210
+    /**
+     * Renders method parameters string
+     */
     private renderMethodParameters(method: spec.Method): string {
+        return this.renderParametersString(method.parameters);
+    }
+
+    /**
+     * Renders parameters string for methods or constructors
+     */
+    private renderParametersString(parameters: spec.Parameter[] | undefined): string {
         const params = [];
-        if (method.parameters) {
-            for (const p of method.parameters) {
-                const isOptionalPrimitive = this.isOptionalPrimitive(p) ? '?' : '';
-                const st = `${this.typeresolver.toDotNetType(p.type)}${isOptionalPrimitive} ${this.nameutils.convertParameterName(p.name)}`;
+        if (parameters) {
+            for (const p of parameters) {
+                let optionalPrimitive = '';
+                let optionalKeyword = '';
+                let type = this.typeresolver.toDotNetType(p.type);
+                if (p.optional) {
+                    if (this.isOptionalPrimitive(p)) {
+                        optionalPrimitive = '?';
+                        optionalKeyword = ' = null';
+                    } else {
+                        optionalKeyword = ' = null';
+                    }
+                }
+                if (p.variadic) {
```

Comment:
is this an else if?

Reply:
It looks like JSII is already doing the "mutual exclusion", but no harm in enforcing it with an if/else

Reply:
Yeah actually **TypeScript** won't allow you to use *both* the variadic syntax and the optional syntax at the same time... So if it's variadic, it cannot be optional, and vice-versa. But there's definitely no harm in checking again.